api-consumption/src/main/java/org/zalando/nakadi/service/subscription/state/StartingState.java

```diff
@@ -41,38 +41,9 @@ private void registerSessionAndStartStreaming() {
         final boolean subscriptionJustInitialized = SubscriptionInitializer.initialize(getZk(),
                 getContext().getSubscription(), getContext().getTimelineService(), getContext().getCursorConverter());
         getContext().getCurrentSpan().setTag("session.id", getContext().getSessionId());
-        if (!subscriptionJustInitialized) {
-            // check if there are streaming slots available
-            final Collection<Session> sessions = getZk().listSessions();
-            final Partition[] partitions = getZk().getTopology().getPartitions();
-            final List<EventTypePartition> requestedPartitions = getContext().getParameters().getPartitions();
-            if (requestedPartitions.isEmpty()) {
-                final long directlyRequestedPartitionsCount = sessions.stream()
-                        .flatMap(s -> s.getRequestedPartitions().stream())
-                        .distinct()
-                        .count();
-                final long autoSlotsCount = partitions.length - directlyRequestedPartitionsCount;
-                final long autoBalanceSessionsCount = sessions.stream()
-                        .filter(s -> s.getRequestedPartitions().isEmpty())
-                        .count();
-                if (autoBalanceSessionsCount >= autoSlotsCount) {
-                    TracingService.logStreamCloseReason(getContext().getCurrentSpan(), "No streaming slots available");
-                    switchState(new CleanupState(new NoStreamingSlotsAvailable(partitions.length)));
-                    return;
-                }
-            }
-
-            // check if the requested partitions are not directly requested by another stream(s)
-            final List<EventTypePartition> conflictPartitions = sessions.stream()
-                    .flatMap(s -> s.getRequestedPartitions().stream())
-                    .filter(requestedPartitions::contains)
-                    .collect(Collectors.toList());
-            if (!conflictPartitions.isEmpty()) {
-                TracingService.logStreamCloseReason(getContext().getCurrentSpan(),
-                        "Partition already taken by other stream of the subscription");
-                switchState(new CleanupState(SubscriptionPartitionConflictException.of(conflictPartitions)));
-                return;
-            }
+        if (!subscriptionJustInitialized &&
```

Comment:
I think that `subscriptionJustInitialized` variable should be removed at all, as it works only once right after first initialization. Also the variable has questionable meaning right now, as it may be returned twice `true` for different concurrent sessions.